clamav/1.0/alpine/scripts/docker-entrypoint-unprivileged.sh

```diff
@@ -4,76 +4,122 @@
-		echo "Updating initial database"
-		
-		# Set "TestDatabases no" and remove "NotifyClamd" for initial download
-		sed -e 's|^\(TestDatabases \)|\#\1|' \
-			-e '$a TestDatabases no' \
-			-e 's|^\(NotifyClamd \)|\#\1|' \
-			/etc/clamav/freshclam.conf > /tmp/freshclam_initial.conf
-		freshclam --foreground --stdout --config-file=/tmp/freshclam_initial.conf
-		rm /tmp/freshclam_initial.conf
-	fi
+# ---------------------------------------------------------------------------
+# default path: launch daemons
+# ---------------------------------------------------------------------------
 
-	if [ "${CLAMAV_NO_FRESHCLAMD:-false}" != "true" ]; then
-		echo "Starting Freshclamd"
-		freshclam \
-		          --checks="${FRESHCLAM_CHECKS:-1}" \
-		          --daemon \
-		          --foreground \
-		          --stdout \
-		          --user="clamav" \
-			  &
-	fi
+# Ensure initial virus database exists, otherwise clamd refuses to start
+echo "[${SCRIPT_FILE}] Updating initial database"
+sed -e 's|^\(TestDatabases \)|#\1|' \
+    -e '$a TestDatabases no' \
+    -e 's|^\(NotifyClamd \)|#\1|' \
+    /etc/clamav/freshclam.conf > /tmp/freshclam_initial.conf
+freshclam --foreground --stdout --config-file=/tmp/freshclam_initial.conf | sed "s/^/[initial-freshclam] /"
+rm /tmp/freshclam_initial.conf
 
-	if [ "${CLAMAV_NO_CLAMD:-false}" != "true" ]; then
-		echo "Starting ClamAV"
-		if [ -S "/tmp/clamd.sock" ]; then
-			unlink "/tmp/clamd.sock"
-		fi
-		clamd --foreground &
-		while [ ! -S "/tmp/clamd.sock" ]; do
-			if [ "${_timeout:=0}" -gt "${CLAMD_STARTUP_TIMEOUT:=1800}" ]; then
-				echo
-				echo "Failed to start clamd"
-				exit 1
-			fi
-			printf "\r%s" "Socket for clamd not found yet, retrying (${_timeout}/${CLAMD_STARTUP_TIMEOUT}) ..."
-			sleep 1
-			_timeout="$((_timeout + 1))"
-		done
-		echo "socket found, clamd started."
-	fi
+# Start clamd (optional, enabled by default)
+if [ "${CLAMAV_NO_CLAMD:-false}" != "true" ]; then
+    echo "[${SCRIPT_FILE}] Starting clamd"
+    [ -S /tmp/clamd.sock ] && unlink /tmp/clamd.sock
 
-	if [ "${CLAMAV_NO_MILTERD:-true}" != "true" ]; then
-		echo "Starting clamav milterd"
-		clamav-milter &
-	fi
+    clamd --foreground > >(sed "s/^/[clamd] /") 2>&1 &
+    clamd_pid=$!
+    child_pids="${child_pids} ${clamd_pid}"
 
-	# Wait forever (or until canceled)
-	exec tail -f "/dev/null"
+    # Wait for socket
+    elapsed=0
+    until [ -S /tmp/clamd.sock ]; do
+        if [ "${elapsed}" -ge "${CLAMD_STARTUP_TIMEOUT}" ]; then
+            echo >&2 "[${SCRIPT_FILE}] Failed to start clamd (socket not found)"
+            kill -TERM "${clamd_pid}" 2>/dev/null || true
+            exit 1
+        fi
+        [ $((elapsed % 5)) -eq 0 ] && \
+            printf "[%s] Waiting for clamd socket... (%s/%s)s\n" "${SCRIPT_FILE}" "${elapsed}" "${CLAMD_STARTUP_TIMEOUT}"
+        sleep 1
+        elapsed=$((elapsed + 1))
+    done
+    echo "[${SCRIPT_FILE}] Socket found after ${elapsed}s, clamd started."
```

Comment:
Checks the condition every 1s, but prints the waiting-message only every 5s, and always prints a new line. Fixes #18